lib/puppet/provider/elasticsearch_plugin/plugin.rb

```diff
@@ -0,0 +1,157 @@
+$LOAD_PATH.unshift(File.join(File.dirname(__FILE__),"..","..",".."))
+
+Puppet::Type.type(:elasticsearch_plugin).provide(:plugin) do
+  desc "A provider for the resource type `elasticsearch_plugin`,
+        which handles plugin installation"
+
+  commands :plugin => '/usr/share/elasticsearch/bin/plugin'
+  commands :es => '/usr/share/elasticsearch/bin/elasticsearch'
```

Comment:
Very small chance they are in a different location, they are meant to be executed as root yeah.